databricks_cli/execution_context/cli.py

```diff
@@ -0,0 +1,251 @@
+        result = client.execute_command(cluster_id, context.id, command, language)
+        if not wait:
+            click.echo(pretty_format(result))
+            return
+
+        command_id = result["id"]
+        _wait_for_command(client, output, cluster_id, context.id, command_id)
+
+
+def _wait_for_command(client, output, cluster_id, context_id, command_id):
+    while True:
+        status = client.get_command_status(cluster_id, context_id, command_id)
+        if not OutputClickType.is_json(output):
+            click.echo("Status: " + status["status"])
+        if status["status"] in ["Cancelled", "Error", "Finished"]:
+            click.echo()
+            _print_command_result(output, status)
+            break
+        time.sleep(1)
+
+
+def _print_command_result(output, status):
+    if OutputClickType.is_json(output):
+        click.echo(pretty_format(status))
+    else:
+        result_type = status["results"]["resultType"]
+        data = status["results"]["data"]
+        if result_type == "text":
+            click.echo("\nCommand ID: %s\n" % status["id"])
+            for line in data.splitlines():
+                click.echo("output > %s" % line)
+        elif result_type == "table":
+            # TODO: add CSV output here
+            headers = [truncate_string(field["name"], 10) for field in status["results"]["schema"]]
+            click.echo(tabulate(tabular_data=data,
+                       headers=headers,
+                       tablefmt='plain', disable_numparse=True))
+        else:
+            click.echo(data)
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help="Cancels a command.")
+@click.option('--cluster-id', required=True, type=ClusterIdClickType(),
+              help=ClusterIdClickType.help)
+@click.option('--context-id', required=True, type=ExecutionContextIdClickType(),
+              help=ExecutionContextIdClickType.help)
+@click.option('--command-id', required=True, type=CommandIdClickType(),
+              help=CommandIdClickType.help)
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def cancel_command_cli(api_client, cluster_id, context_id, command_id): #  NOQA
+    """
+    Cancels a command.
+    """
+    ExecutionContextApi(api_client).cancel_command(cluster_id, context_id, command_id)
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help="Gets information about a command.")
+@click.option('--cluster-id', required=True, type=ClusterIdClickType(),
+              help=ClusterIdClickType.help)
+@click.option('--context-id', required=True, type=ExecutionContextIdClickType(),
+              help=ExecutionContextIdClickType.help)
+@click.option('--command-id', required=True, type=CommandIdClickType(),
+              help=CommandIdClickType.help)
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def get_command_status_cli(api_client, cluster_id, context_id, command_id): #  NOQA
```

Comment:
this command makes no sense in isolation within the CLI from end-user perspective.

Reply:
It does make sense when you start a command async.

Reply:
why people need to start individual notebook command async if they have run-submit jobs, that also have a snapshot of notebook being executed?